src/index.js

```diff
@@ -0,0 +1,306 @@
+const state = {
+  playgroundTemp: 60,
+  displayedLiveTemp: null,
+  liveTempInK: null,
+  liveWeather: null,
+  unit: 'F',
+  city: 'Seattle',
+};
+
+// --- HTML DOM Elements ---
+const playgroundTempElement = document.getElementById('playground-temp');
+const skyOptionElement = document.getElementById('sky-select');
+const celsiusButton = document.getElementById('celsius');
+const fahrenheitButton = document.getElementById('fahrenheit');
+const inputElement = document.querySelector('input');
+
+// Other global const
+const degreeSymbol = String.fromCharCode(176);
+
+// --- City Name --- //
+const displayCityName = () => {
+  document.getElementById('city-name').textContent = state.city;
+};
+
+const resetCityName = () => {
+  state.city = 'Seattle';
+  displayCityName();
+};
+
+const submitCity = (event) => {
+  // update city name from input form
+  if (inputElement.value) {
+    state.city = inputElement.value;
+  }
+  inputElement.value = '';
+  event.preventDefault();
+  displayCityName();
+};
+
+const cityName = document.getElementById('city-name');
+
+const updateCityName = (e) => {
+  cityName.textContent = e.target.value;
+  if (e.target.value.length === 0) {
+    resetCityName();
+  }
+};
+
+const getCurrentCity = () => {
+  // using HTML Geolocation API to get coordinates of the current location if browser permits
+  const success = async (position) => {
+    // This function will make an API call to LocationIQ reverse geocoding API
+    // to get city name from lat & lon
+    // and display the current city name
+    console.log('HTML Geolocation API called.');
+    const lat = position.coords.latitude;
+    const lon = position.coords.longitude;
+    const city = await getCityFromLatLon(lat, lon);
+    if (city) {
+      state.city = `${city}`;
+    }
+    displayCityName();
+    await displayWeatherInfo(); // API call
+  };
+
+  const error = (err) => {
+    console.warn(`ERROR(${err.code}):${err.message}`);
+    alert('Your browser does not permit location sharing.');
+  };
+
+  navigator.geolocation.getCurrentPosition(success, error);
+};
+
+const getCityFromLatLon = async (lat, lon) => {
+  // This function will make an API call to LocationIQ reverse geocoding API
+  try {
+    const response = await axios.get(
+      `https://weather-report-backend.herokuapp.com/location/reverse?lat=${lat}&lon=${lon}`
+    );
+    const city = response.data.address.city;
+    console.log('LocationIQ Reverse Geocoding called by getCityFromLatLon');
+    return city;
+  } catch (error) {
+    console.log(error);
+    console.log('error in getCityFromLatLon');
+  }
+};
+
+// --- Playground box --- //
+
+const updatePlaygroundColor = (tempTextColor, gardenBgColor, landscapeText) => {
+  const weatherGardenElement = document.getElementById('weather-garden');
+  const landscapeElement = document.getElementById('landscape');
+
+  playgroundTempElement.style.color = tempTextColor;
+  weatherGardenElement.style.backgroundColor = gardenBgColor;
```

Comment:
It would have been nice if these used CSS classes (say some named `hot`, `warm`, etc) since that way the styling would all be in one place and you could change color schemes more easily.  (For example if you wanted a red to purple to blue temperature gradient.)